pam/integration-tests/gdm-integration_test.go

```diff
@@ -0,0 +1,471 @@
+			wantPamErrorMessages: []string{
+				"Sending GDM event failed: Conversation error",
+			},
+			wantError:       pam.ErrSystem,
+			wantAcctMgmtErr: ignoreError,
+		},
+		"Error (ignored) on local broker causes fallback error": {
+			pamUser:    "user-foo",
+			brokerName: localBrokerName,
+			wantPamInfoMessages: []string{
+				"auth=" + pamDebugIgnoreError,
+			},
+			wantError:       ignoreError,
+			wantAcctMgmtErr: pam.ErrAbort,
+		},
+		"Error on authenticating user2 with too many retries": {
+			pamUser: "user2",
+			authModeIDs: []string{
+				passwordAuthID,
+				passwordAuthID,
+				passwordAuthID,
+				passwordAuthID,
+				passwordAuthID,
+				passwordAuthID,
+			},
+			eventPollResponses: map[gdm.EventType][]*gdm.EventData{
+				gdm.EventType_startAuthentication: {
+					gdm_test.IsAuthenticatedEvent(&authd.IARequest_AuthenticationData_Challenge{
+						Challenge: "not goodpass",
+					}),
+					gdm_test.IsAuthenticatedEvent(&authd.IARequest_AuthenticationData_Challenge{
+						Challenge: "another not goodpass",
+					}),
+					gdm_test.IsAuthenticatedEvent(&authd.IARequest_AuthenticationData_Challenge{
+						Challenge: "even more not goodpass",
+					}),
+					gdm_test.IsAuthenticatedEvent(&authd.IARequest_AuthenticationData_Challenge{
+						Challenge: "not yet goodpass",
+					}),
+					gdm_test.IsAuthenticatedEvent(&authd.IARequest_AuthenticationData_Challenge{
+						Challenge: "really, it's not a goodpass!",
+					}),
+					gdm_test.IsAuthenticatedEvent(&authd.IARequest_AuthenticationData_Challenge{
+						Challenge: "goodpass",
+					}),
+				},
+			},
+			wantPamErrorMessages: []string{
+				"invalid password 'really, it's not a goodpass!', should be 'goodpass'",
+			},
+			wantError:       pam.ErrAuth,
+			wantAcctMgmtErr: ignoreError,
+		},
+		"Error on authenticating unknown user": {
+			pamUser: "user-unknown",
+			eventPollResponses: map[gdm.EventType][]*gdm.EventData{
+				gdm.EventType_startAuthentication: {
+					gdm_test.IsAuthenticatedEvent(&authd.IARequest_AuthenticationData_Challenge{
+						Challenge: "",
+					}),
+				},
+			},
+			wantPamErrorMessages: []string{
+				"user not found",
+			},
+			wantError:       pam.ErrAuth,
+			wantAcctMgmtErr: ignoreError,
+		},
+		"Error on invalid fido ack": {
+			pamUser:     "user-mfa",
+			authModeIDs: []string{passwordAuthID, fido1AuthID},
+			eventPollResponses: map[gdm.EventType][]*gdm.EventData{
+				gdm.EventType_startAuthentication: {
+					gdm_test.IsAuthenticatedEvent(&authd.IARequest_AuthenticationData_Challenge{
+						Challenge: "goodpass",
+					}),
+					gdm_test.IsAuthenticatedEvent(&authd.IARequest_AuthenticationData_Wait{}),
+				},
+			},
+			wantPamErrorMessages: []string{
+				fido1AuthID + " should have wait set to true",
+			},
+			wantError:       pam.ErrAuth,
+			wantAcctMgmtErr: ignoreError,
+		},
+	}
+	for name, tc := range testCases {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Cleanup(pam_test.MaybeDoLeakCheck)
+			t.Parallel()
```

Comment:
any reason to have `t.Parallel()` after the cleanup registration?

Reply:
Just a missed thing